src/Microsoft.AspNetCore.Mvc.Razor.Host/ViewHierarchyUtility.cs

```diff
@@ -53,39 +53,53 @@ public static IEnumerable<string> GetViewStartLocations(string applicationRelati
         /// </remarks>
         public static IEnumerable<string> GetViewImportsLocations(string applicationRelativePath)
         {
-            return GetHierarchicalPath(applicationRelativePath, ViewImportsFileName);
+            return GetHierarchicalLocations(applicationRelativePath, ViewImportsFileName);
         }
 
-        private static IEnumerable<string> GetHierarchicalPath(string relativePath, string fileName)
+        /// <summary>
+        /// Gets the locations for <paramref name="fileName"/>s that are applicable to the specified path.
+        /// </summary>
+        /// <param name="applicationRelativePath">The application relative path of the file to locate
+        /// files named <paramref name="fileName"/>.</param>
+        /// <param name="fileName">The file name.</param>
+        /// <returns>A sequence of paths that represent potential <c>_ViewImports</c> locations.</returns>
+        /// <remarks>
+        /// This method returns paths starting from the directory of <paramref name="applicationRelativePath"/> and
+        /// moves upwards until it hits the application root.
+        /// e.g.
+        /// /Views/Home/View.cshtml -> [ /Views/Home/FileName.cshtml, /Views/FileName.cshtml,
+        ///                              /FileName.cshtml ]
+        /// </remarks>
+        public static IEnumerable<string> GetHierarchicalLocations(string applicationRelativePath, string fileName)
```

Comment:
Moving this to RazorProject as follow up (https://github.com/aspnet/Razor/issues/962)